src/decouplededitor.js

```diff
@@ -107,18 +133,107 @@ export default class DecoupledEditor extends Editor {
 	destroy() {
 		// Cache the data, then destroy.
 		// It's safe to assume that the model->view conversion will not work after super.destroy().
-		const data = this.getData();
+		const data = [];
+		for ( const rootName of this._rootNames ) {
+			data.push( this.getData( rootName ) );
+		}
 
 		this.ui.destroy();
 
 		return super.destroy()
 			.then( () => {
-				if ( this.sourceElement ) {
-					setDataInElement( this.sourceElement, data );
+				if ( this._sourceElements.length ) {
+					for ( let i = 0; i < this._rootNames.length; i++ ) {
+						setDataInElement( this._sourceElements[ i ], data[ i ] );
+					}
 				}
 			} );
 	}
 
+	/**
+	 * Gets the data from the editor.
+	 *
+	 *		editor.getData(); // -> '<p>This is editor!</p>'
+	 *		editor.getData( 'rootName' ); // -> '<h1>CKEditor 5</h1>'
+	 *
+	 * By default the editor outputs HTML. This can be controlled by injecting a different data processor.
+	 * See the {@glink features/markdown Markdown output} guide for more details.
+	 *
+	 * Note: Not only is the format of the data configurable, but the type of the `getData()`'s return value does not
+	 * have to be a string either. You can e.g. return an object or a DOM `DocumentFragment`  if you consider this
+	 * the right format for you.
+	 *
+	 * @method #getData
+	 * @param {String} [rootName='main'] The root name from which to get data.
+	 * @returns {String} Output data.
+	 */
+	getData( rootName = 'main' ) {
+		if ( !this._rootNames.includes( rootName ) ) {
+			/**
+			 * Thrown when there is an attempt to get data on a non-existing root.
+			 *
+			 * @error trying-to-get-data-on-non-existing-root
+			 */
+			throw new CKEditorError( 'trying-to-get-data-on-non-existing-root: ' +
```

Comment:
`editor-getData-non-existent-root`

It's an id, not a sentence – it should look like a namespace – from general to specific information.